### PR TITLE
fix: support remove unpartitioned same name cookie first

### PR DESCRIPTION
**Pull request by fengmk2**

### CODE REVIEW

File: test/lib/cookies.test.js [hunk 1]
```diff
@@ -513,5 +513,28 @@ describe('test/lib/cookies.test.js', () => {
...
+      };
+      cookies.set('foo', 'hello', opts);
+
+      assert(opts.signed === 1);
+      assert(opts.secure === undefined);
+      const headers = cookies.ctx.response.headers['set-cookie'];
+      // console.log(headers);
+      assert.equal(headers.length, 4);
+      assert.equal(headers[0], 'foo=; path=/; expires=Thu, 01 Jan 1970 00:00:00 GMT; httponly');
+      assert.equal(headers[1], 'foo.sig=; path=/; expires=Thu, 01 Jan 1970 00:00:00 GMT; httponly');
```
Comment (fengmk2) [inline test/lib/cookies.test.js:535]: 有问题，少了 secure

**Review by killagu — APPROVED** [review]
LGTM

**Comment by DorothyGuan** [comment]
index.d.ts的removeUnpartitioned定义没加

**Comment by DorothyGuan** [comment]
overwrite属性对removeUnpartitioned也有影响，如果overwrite是false，removeUnpartitioned并不会生效

**Comment by fengmk2** [comment]
> overwrite属性对removeUnpartitioned也有影响

已修复

**Pull request merged**